### PR TITLE
doc/cephfs/posix: put posix notes in perspective

### PR DESCRIPTION
**Pull request by liewegas**
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: doc/cephfs/posix.rst [hunk 1]
```diff
@@ -47,3 +47,36 @@ POSIX semantics for various reasons:
...
+
+Perspective
+-----------
+
+People talk a lot about "POSIX compliance," but in reality most file
+system implementations do not strictly adhere to the spec, including
+local Linux file systems like ext4 and XFS.  For example, for
+performance reasons, the atomicity requirements for reads are relaxed:
+processing reading from a file that is also being written may see torn
+results.
```
Comment (liewegas) [inline doc/cephfs/posix.rst]: @ukernel is this accurate?  are there other local fs diversions from posix worth noting?

File: doc/cephfs/posix.rst [hunk 2]
```diff
@@ -47,3 +47,43 @@ POSIX semantics for various reasons:
...
+
+Similarly, NFS has extremely weak consistency semantics when multiple
+clients are interacting with the same files or directories, opting
+instead for "close-to-open".  In the world of network attached
+storage, where most environments use NFS, whether or not the server's
+file system is "fully POSIX" is may not relevant, and whether client
```
Comment (jdurgin) [inline doc/cephfs/posix.rst]: typo: extra 'may'

File: doc/cephfs/posix.rst (hunk 1 above)
Comment (ukernel) [reply doc/cephfs/posix.rst]: yes, it's accurate. I don't think of any other diversions

**Comment by liewegas** [comment]
@batrick this look ok to you?

**Pull request merged**